raft/rawnode.go

```diff
@@ -103,9 +103,14 @@ func NewRawNode(config *Config, peers []Peer) (*RawNode, error) {
 			r.addNode(peer.ID)
 		}
 	}
+
 	// Set the initial hard and soft states after performing all initialization.
 	rn.prevSoftSt = r.softState()
-	rn.prevHardSt = r.hardState()
+	if lastIndex == 0 {
```

Comment:
@bdarnell I try to fix the rawnode impl here. We still need to persist the initial hard state if raft is node recovered from an existing state.